meson.build

```diff
@@ -0,0 +1,392 @@
+  shlext = '.so'
+  exeext = ''
+endif
+
+conf.set_quoted('SHLEXT', shlext)
+conf.set_quoted('EXEEXT', exeext)
+
+if get_option('debug')
+  conf.set('WITH_DEBUG', 1)
+  conf.set('_DEBUG', 1)
+else
+  conf.set('NDEBUG', 1)
+endif
+
+if get_option('strict')
+  conf.set('WITH_STRICT', 1)
+endif
+
+prefix = get_option('prefix')
+datadir = join_paths(prefix, get_option('datadir'))
+bindir = join_paths(prefix, get_option('bindir'))
+libdir = join_paths(prefix, get_option('libdir'))
+libexecdir = join_paths(prefix, get_option('libexecdir'))
+sysconfdir = join_paths(prefix, get_option('sysconfdir'))
+pkgdatadir = join_paths(datadir, meson.project_name())
+privatedir = join_paths(libexecdir, meson.project_name())
+
+common_c_args = [
+  '-DBINDIR="@0@"'.format(bindir),
+  '-DPRIVATEDIR="@0@"'.format(privatedir),
+  '-DSYSCONFDIR="@0@"'.format(sysconfdir)
+]
+
+top_source_dir = meson.current_source_dir()
+top_build_dir = meson.current_build_dir()
+
+tests_c_args = [
+  '-DSRCDIR="@0@"'.format(top_source_dir),
+  '-DBUILDDIR="@0@"'.format(top_build_dir)
+]
+
+conf.set('SIZEOF_UNSIGNED_LONG', cc.sizeof('unsigned long'))
+
+if host_system != 'windows'
+  pthread_create_deps = []
+  if not cc.has_function('pthread_create')
+    libpthread = cc.find_library('pthread', required: false)
+    if cc.has_function('pthread_create', dependencies: libpthread)
+      pthread_create_deps += libpthread
+    else
+      error('could not find pthread_create')
+    endif
+  endif
+
+  if not cc.has_function('nanosleep')
+    nanosleep_deps = cc.find_library('rt', required: false)
+    if not cc.has_function('nanosleep', dependencies: nanosleep_deps)
+      error('could not find nanosleep')
+    endif
+  endif
+
+  libdl_deps = []
+  if not cc.has_function('dlopen')
+    libdl = cc.find_library('dl', required: false)
+    if cc.has_function('dlopen', dependencies: libdl)
+      libdl_deps += libdl
+    else
+      error('could not find dlopen')
+    endif
+  endif
+
+  # for Solaris we need -lsocket -lnsl for socket stuff, gethostbyname
+  # is just a dummy to find -lnsl
+  libnsl = cc.find_library('nsl', required: false)
+  if libnsl.found()
+    cc.has_function('gethostbyname', dependencies: libnsl)
+  endif
+
+  libsocket = cc.find_library('socket', required: false)
+  if libsocket.found()
+    if not cc.has_function('connect', dependencies: libsocket)
+      error('could not find socket')
+    endif
+  endif
+
+  if cc.has_header('locale.h')
+    conf.set('HAVE_LOCALE_H', 1)
+    if cc.has_header_symbol('locale.h', 'locale_t')
+      conf.set('HAVE_LOCALE_T', 1)
+      if cc.has_function('newlocale', prefix: '#include <locale.h>')
+        conf.set('HAVE_NEWLOCALE', 1)
+      endif
+      if cc.has_function('strerror_l', prefix: '#include <string.h>')
+        conf.set('HAVE_STRERROR_L', 1)
+      endif
+    endif
+  endif
+
+  # These are things we can work around
+  headers = [
+    'sys/resource.h',
+    'ucred.h'
+  ]
+
+  foreach h : headers
+    if cc.has_header(h)
```

Comment:
Hmm, this seems not that simple, because the C code assumes the `HAVE_FOO_H` is undefined (not set to 0).

Reply:
ok! I've rechecked it, my fault. In this case `set` method should be used instead of `set10`, so meson doesn't define it at all, which is perfect for `ifdef`.